.github/workflows/test-pygenesys.yml

```
@@ -0,0 +1,66 @@
+# Preamble
+name: Test Pygenesys
+
+on:
+  push:
+    paths:
+      - 'pygenesys/**'
+      - 'pygenesys/tests/**'
+  pull_request:
+  # enable worflow to be run manually
+  workflow_dispatch:
+
+env:
+  CACHE_NUMBER: 0 #change to manually reset cache
+
+jobs:
+  test-pygenesys:
+    runs-on: ubuntu-latest
```

```suggestion
    runs-on: ${{ matrix.os }}
    strategy:
       matrix: [ubuntu-latest, windows-latest, macOS-latest]
```

I think this should do it.

---

We'll want to have a cache for each os.

---

@yardasol the windows build fails because windows doesn't support the default solver (CBC) for temoa. I don't think our CI should include a temoa build since we only have unit tests. We can introduce a separate check if/when integration tests are introduced. Since `pygenesys` is separate from temoa, the latter could make breaking changes for their project which would show up as failed checks for this repo -- which isn't our job to handle imo. Thoughts?

---

I think that's reasonable.